test/TestImportIncludeRole.py

```diff
@@ -55,56 +56,67 @@
-        results = runner.run()
-        assert 'only when shell functionality is required' in str(results)
-        assert 'All tasks should be named' in str(results)
-
-    def test_include_role_inline_args(self):
-        fh = self._get_play_file(PLAY_INCLUDE_ROLE_INLINE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        results = runner.run()
-        assert 'only when shell functionality is required' in str(results)
-        assert 'All tasks should be named' in str(results)
-
-    def test_invalid_import_role(self):
-        fh = self._get_play_file(PLAY_IMPORT_ROLE_INCOMPLETE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        with pytest.raises(RuntimeError, match="Failed to find required 'name' key in import_role"):
-            runner.run()
+def _setup_play(base_dir, playbook_text):
+    role_tasks_dir = base_dir / 'test-role' / 'tasks'
+    role_tasks_dir.mkdir(parents=True)
+    (role_tasks_dir / 'main.yml').write_text(ROLE_TASKS_MAIN)
+    (role_tasks_dir / 'world.yml').write_text(ROLE_TASKS_WORLD)
+    play_path = base_dir / 'playbook.yml'
+    play_path.write_text(playbook_text)
+    return str(play_path)
+
+
+def ids(value):
+    return {PLAY_IMPORT_ROLE: 'IMPORT_ROLE',
+            PLAY_IMPORT_ROLE_INLINE: 'IMPORT_ROLE_INLINE',
+            PLAY_INCLUDE_ROLE: 'INCLUDE_ROLE',
+            PLAY_INCLUDE_ROLE_INLINE: 'INCLUDE_ROLE_INLINE',
+            PLAY_IMPORT_ROLE_INCOMPLETE: 'IMPORT_ROLE_INCOMPLETE',
+            }[value[0]]
+
+
+@pytest.fixture(params=(
+    PlayAndMessages(PLAY_IMPORT_ROLE,
+                    ['only when shell functionality is required'],
+                    ),
+    PlayAndMessages(PLAY_IMPORT_ROLE_INLINE,
+                    ['only when shell functionality is require'],
+                    ),
+    PlayAndMessages(PLAY_INCLUDE_ROLE,
+                    ['only when shell functionality is require',
+                     'All tasks should be named'],
+                    ),
+    PlayAndMessages(PLAY_INCLUDE_ROLE_INLINE,
+                    ['only when shell functionality is require',
+                     'All tasks should be named'],
+                    ),
+), ids=ids)
+def play_and_messages(request, tmp_path):
+    playbook_text, messages = request.param
+    return _setup_play(tmp_path, playbook_text), messages
+
+
+@pytest.fixture(params=[
+    PlayAndMessages(PLAY_IMPORT_ROLE_INCOMPLETE,
+                    "Failed to find required 'name' key in import_role"),
+], ids=ids)
+def play_and_error(request, tmp_path):
+    playbook_text, error = request.param
+    return _setup_play(tmp_path, playbook_text), error
+
+
+def test_import_role(default_rules_collection, play_and_messages):
+    playbook, messages = play_and_messages
+    runner = Runner(default_rules_collection, playbook, [], [], [])
+
+    results = runner.run()
+
+    for message in messages:
+        assert message in str(results)
+
+
+def test_invalid_import_role(default_rules_collection, play_and_error):
```

Comment:
It'd be easier to follow if parametrize was applied here and not hidden in the fixture implementation or maybe if there were fixtures decoupling playbook&error back into two args

Reply:
I think I found a better way.